tests/unit/s2n_self_talk_ktls_test.c

```diff
@@ -61,8 +61,11 @@ static S2N_RESULT s2n_setup_connections(struct s2n_connection *server,
 /* Unlike our other self-talk tests, this test cannot use AF_UNIX / AF_LOCAL.
  * For a real self-talk test we need real kernel support for kTLS, and only
  * AF_INET sockets support kTLS.
+ * 
+ * Pass in file as an argument, so that we can close it once the child process
+ * exits.
  */
-static S2N_RESULT s2n_new_inet_socket_pair(struct s2n_test_io_pair *io_pair)
+static S2N_RESULT s2n_new_inet_socket_pair(struct s2n_test_io_pair *io_pair, int file)
```

Comment:
> I'm not sure that closing an arbitrary file descriptor should be the responsibility of this function.

I did realize that this is not the original purpose of the function. Since there are not too many places using the file, my current solution is to open and close it for each file usage.

Reply:
Fix this issue in this [comment](https://github.com/aws/s2n-tls/pull/4852#discussion_r1823145707).